src/Metadata/Resource/Factory/DirectoryResourceMetadataFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractorInterface;
+
+final class DirectoryResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    private $propertyInfo;
+    private $defaults;
+
+    public function __construct(array $defaults = [], PropertyInfoExtractorInterface $propertyInfo)
+    {
+        $this->propertyInfo = $propertyInfo;
+        $this->defaults = $defaults + ['attributes' => []];
+    }
+
+    public function create(string $resourceClass): ResourceMetadata
+    {
+        return $this->createMetadata(new \ReflectionClass($resourceClass));
+    }
+
+    private function createMetadata(
+        \ReflectionClass $reflectionClass
+    ): ResourceMetadata {
+        $properties = array_map(function (\ReflectionProperty $reflectionProperty) {
+            return $reflectionProperty->getName();
+        }, $reflectionClass->getProperties());
+        $attributes = (null === $properties && [] === $this->defaults['attributes']) ? null :
+            (array)
+            $properties + $this->defaults['attributes'];
+
+        return new ResourceMetadata(
+            $reflectionClass->getShortName(),
```

Comment:
Isn't that the responsibility of https://github.com/api-platform/core/blob/2f941fc3c0dc9dd257da3b34a6bb3f5c21c4230c/src/Metadata/Resource/Factory/ShortNameResourceMetadataFactory.php#L23 ?